src/platform_impl/macos/event_loop.rs

```diff
@@ -254,16 +229,14 @@ impl<T> EventLoop<T> {
         let panic_info: Rc<PanicInfo> = Default::default();
         setup_control_flow_observers(mtm, Rc::downgrade(&panic_info));
 
-        let (sender, receiver) = mpsc::channel();
         Ok(EventLoop {
             app,
             delegate: delegate.clone(),
-            sender,
-            receiver: Rc::new(receiver),
             window_target: RootWindowTarget {
                 p: ActiveEventLoop { delegate, mtm },
                 _marker: PhantomData,
             },
+            user_wake_up: Arc::new(AtomicBool::new(false)),
```

Comment:
You're creating `user_wake_up` twice in this function?

```suggestion
            user_wake_up,
```